parser_library/src/parsing/grammar/macro_operand_rules.g4

```diff
@@ -53,58 +53,63 @@ mac_preproc[int* paren_count]
 		(
 			ORDSYMBOL (ORDSYMBOL|NUM)*
 			|
-			LPAR {++*$paren_count;}
+			LPAR
 			|
 			AMPERSAND
 		)
 		|
-		LPAR {++*$paren_count;}
+		LPAR
 		|
-		RPAR {--*$paren_count;}
+		RPAR
 		|
 		APOSTROPHE
 		(~(APOSTROPHE|ATTR|CONTINUATION))*
 		(APOSTROPHE|ATTR)
 		|
 		ATTR
 		(
-			{!is_previous_attribute_consuming(*$paren_count == 0, _input->LT(-2))}?
-			(~(APOSTROPHE|ATTR|CONTINUATION))*
-			(APOSTROPHE|ATTR)
-		)?
+			{is_attribute_consuming(_input->LT(-2))}?
+			(
+				ORDSYMBOL
+				|
+				EQUALS
+				ORDSYMBOL
+				|
+				(ASTERISK|MINUS|PLUS|LT|GT|SLASH|VERTICAL|IDENTIFIER|NUM|DOT|LPAR|RPAR)
+				(~(APOSTROPHE|ATTR|CONTINUATION)|(APOSTROPHE|ATTR)(APOSTROPHE|ATTR))*
```

Comment:
I don't quite follow this change... if the attribute is consuming then a string was not started and everything should be picked up "in the next round" by the other rules, shouldn't it?

Reply:
The main reason for this change was that I wanted to be able to diagnose "missing apostrophe" specifically and as soon as possible (ideally here in `mac_preproc`).

The best I could do here without compromising much of the diagnostics is the following:
```suggestion
			{is_attribute_consuming(_input->LT(-2))}?
			AMPERSAND
			ORDSYMBOL
			(ORDSYMBOL|NUM)*
			(APOSTROPHE|ATTR)?
			|
			(
				{!is_attribute_consuming(_input->LT(-2))}?
				|
				{is_attribute_consuming(_input->LT(-2))}?
				{!can_attribute_consume(_input->LT(1))}?
			)
			(
				(~(APOSTROPHE|ATTR|CONTINUATION)|(APOSTROPHE|ATTR)(APOSTROPHE|ATTR))*
				(APOSTROPHE|ATTR)
			)
		)?
```

If you want I can put it into the next commit

Reply:
Isn't that a completely different grammar?

The original rule was parsing a string, the new one also parses things that are not valid strings

Reply:
Well, variable symbols can be analyzed only after their substitution with an earlier defined value. 

E.g. this is valid 
```
&VAR     SETC 'J'
         MAC (I'&VAR)
```

while this is not
```
&VAR     SETC '9'
         MAC (I'&VAR)
```

Reply:
then it looks like we need `{!is_attribute_consuming(_input->LT(-2)) || !can_be_consumed(_input->LT(1)) }?` predicate?

Reply:
Yep, and this part too
```
			{is_attribute_consuming(_input->LT(-2))}?
			AMPERSAND
			ORDSYMBOL
			(ORDSYMBOL|NUM)*
			(APOSTROPHE|ATTR)?
```